File: src/LIT.Smabu.UseCases/Catalogs/CatalogDTO.cs
using LIT.Smabu.Domain.Base;
using LIT.Smabu.Domain.CatalogAggregate;
using LIT.Smabu.UseCases.Base;

namespace LIT.Smabu.UseCases.Catalogs
{
    public record CatalogDTO(CatalogId Id, string DisplayName, string Name, CatalogGroupDTO[] Groups) : IDTO
    {
        internal static Result<CatalogDTO> Create(Catalog catalog)
        {
            CatalogGroupDTO[] groups = catalog.Groups.Select(CatalogGroupDTO.Create).ToArray();
            return new CatalogDTO(catalog.Id, catalog.DisplayName, catalog.Name, groups);
        }
    }
}

File: src/lit.smabu.react/src/types/domain/catalog-dto.ts

import { CatalogGroupDTO } from './catalog-group-dto';
import { CatalogId } from './catalog-id';

export interface CatalogDTO {
  id?: CatalogId;
  name?: string | null;
  groups?: Array<CatalogGroupDTO> | null;
  displayName?: string | null;
}
